### PR TITLE
currency: fiat_provider config option takes precedence

### PR DESCRIPTION
**Pull request by Exirel**
### Description

Tin, part of #2120.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/modules/currency.py [hunk 1]
```diff
@@ -215,28 +216,25 @@ def update_rates(bot):
     rates_crypto = response.json()
 
     # Update fiat rates
-    if bot.config.currency.fixer_io_key is not None:
+    settings = bot.settings.currency
```
Comment (dgw) [inline sopel/modules/currency.py:219]: I have mixed feelings about making such aliases. To someone unfamiliar with the structure of a Sopel instance, reading core plugin code to learn from existing examples, this is easy to miss (and then later wonder where the hell `settings` comes from).

In this particular case, converting to use this alias also pollutes the diff 😝

**Review by dgw — APPROVED** [review]
I'm nitpicking. There's nothing actually "wrong" with the patch; it just took a bit more mental energy to review than it would have otherwise due to extraneous code style changes. :P

File: sopel/modules/currency.py (hunk 1 above)
Comment (Exirel) [reply sopel/modules/currency.py:219]: Yeah, I'll change that to something easier to read, even for me this is not optimal.

**Comment by Exirel** [comment]
And done.

**Review by dgw — APPROVED** [review]
Nitpick resolved, I see. Squash away 🚀

**Pull request merged**